src/language/worker.ts

```diff
@@ -3,21 +3,39 @@ import {
   MongoClientOptions
 } from '@mongosh/service-provider-server';
 import { CompletionItemKind } from 'vscode-languageserver/node';
-import { EJSON } from 'bson';
+import { EJSON, Document } from 'bson';
 import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
 import parseSchema = require('mongodb-schema');
 import { parentPort, workerData } from 'worker_threads';
-
-import type { PlaygroundResult, PlaygroundDebug, ShellExecuteAllResult } from '../types/playgroundType';
+import { PlaygroundResult, PlaygroundDebug, ShellExecuteAllResult } from '../types/playgroundType';
 import { ServerCommands } from './serverCommands';
 
-type EvaluationResult = {
+interface EvaluationResult {
   printable: any;
   type: string | null;
-};
+}
+
 type WorkerResult = ShellExecuteAllResult;
 type WorkerError = any | null;
 
+const content = ({ type, printable }: EvaluationResult) => {
+  if (type === 'Cursor' || type === 'AggregationCursor') {
+    return JSON.parse(EJSON.stringify(printable.documents));
+  }
+
+  return typeof printable === 'string'
+    ? printable
+    : JSON.parse(EJSON.stringify(printable));
+};
+
+const language = (evaluationResult: EvaluationResult) => {
+  if (typeof content(evaluationResult) === 'object' && content(evaluationResult) !== null) {
```

Comment:
This is pretty expensive, evaluating `content()` twice just to check the type of the result. Is that something that we can avoid? Either by making this check the type explicitly or at least storing `content(evaluationResult)` once in a variable and then using that here.